# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/media/webaudio/MediaBufferDecoder.cpp
# Commit: d5e5afb3ea89
# Full Hash: d5e5afb3ea89fdf5af84a9b2c3abc809f2f8f33e
# Author: az <azebrowski@mozilla.com>
# Date: 2025-06-25 03:43:13
# Description:
#   Bug 1971539 - Exit early on failure to add channel data when finishing audio decoding. r=padenot
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D254762
# ==============================================================================

diff -r ad8972939a88 -r d5e5afb3ea89 dom/media/webaudio/MediaBufferDecoder.cpp
--- a/dom/media/webaudio/MediaBufferDecoder.cpp	Tue Jun 24 14:02:37 2025 +0000
+++ b/dom/media/webaudio/MediaBufferDecoder.cpp	Tue Jun 24 14:07:36 2025 +0000
@@ -551,6 +551,7 @@
       newBuffers = CreateChannelBuffers(audioData->mChannels, resampledFrames);
       if (!newBuffers) {
         ReportFailureOnMainThread(WebAudioDecodeJob::UnknownError);
+        return;
       }
       RefPtr<ThreadSharedFloatArrayBufferList> oldBuffers =
           mDecodeJob.mBuffer.mBuffer->AsThreadSharedFloatArrayBufferList();
